### ISSUE TITLE
Add support for Opentofu for language server

### ISSUE THREAD

**Issue by marguslaanem** [zed-industries/zed#10373]
### Check for existing issues

- [X] Completed

### Describe the feature

At the moment language server requires terraform binary to exist to autoformat document. I wish it would be able to detect Opentofu  also. Or atleast I wish there was an option to change from terraform to opentofu.

### If applicable, add mockups / screenshots to help present your vision of the feature

The error is:
Formatting failed: failed to format via language server. Click to see logs.

Caused by:
    Terraform (CLI) is required. Please install Terraform or make it available in $PATH

**Comment by JosephTLyons** [comment]
Hey @marguslaanem - I imagine the terraform language will be pulled out into an installable extension at some point, where you can decide to simply not install it. At that point, a community member can make an `opentofu` extension that users can choose to use.

**Comment by marguslaanem** [comment]
If that means that HCL language autoformat on save functionality will move from zed to and extension then I guess its fine. I can use a workaround until then.

**Comment by taras-mrtn** [comment]
[Here](https://github.com/zed-industries/zed/commit/b5b872656bb92d989043e388f9907ee582c99dea) Terraform was extracted to the extension 
Can this issue be closed?

**Comment by marguslaanem** [comment]
I will close the issue and hope someone makes an opentofu extension as well.

**Comment by taras-mrtn** [comment]
@marguslaanem OpenTofu doesn't have its own language server. I already checked and figured out they are working on this. Some guy forked terraform-ts to make it opentofu-ls, but it is not under the OpenTofu organization, this is his own fork. OpenTofu in some issue said, that they are communicating regarding language server, so once it is ready and official - I think I will be able to work on Zed extension for it.

**Comment by ashpool37** [comment]
I've adapted the Terraform extension to work with the [recently released](https://opentofu.org/blog/opentofu-1-10-0/#ecosystem-maturity) OpenTofu language server ([`tofu-ls`](https://github.com/opentofu/tofu-ls)):

https://github.com/ashpool37/zed-extension-opentofu

I didn't change any logic except making it download the language server binary directly from GitHub since they don't publish them on their website. Other than that, all changes are purely branding-related: I replaced Terraform mentions with OpenTofu where appropriate.

Try running the extension and let me know if you run into anything. Works for me so far, and soon I might open a PR to get it listed on the Zed extensions registry.

---

Update: the extension is now published and available to anyone running Zed 0.192.x or newer.
